source/fundamentals/typescript.txt

```diff
@@ -79,81 +84,119 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
 class in the
 :ref:`Find Multiple Documents Usage Example <node-driver-find-usage-example-code-snippet>`.
 
-Limitations
------------
+To view an example of a mutually recursive type, which is not supported by the
+preceding classes, see the :ref:`<node-driver-limitations-mutual-recursion>` section.
 
-.. _node-driver-typescript-limitations-dot-notation:
+.. _node-driver-limitations:
 
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
+Limitations For Driver Version {+version+}
+----------------------------------
 
-.. tabs::
+The following subsections describe the TypeScript specific limitations of
+version {+version+} of the {+driver-long+}. 
 
-  .. tab:: Dot Notation
-     :tabid: dot-notation
+Many limitations of the {+driver-short+} relate to **recursive types**.
+A recursive type is a type that references itself. You can update
+the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be recursive by allowing a pet to have its own pet. The following is the
+recursive ``Pet`` interface:
 
-     The following code snippet does not raise a type error:
+.. _node-driver-limitations-recursive-pet:
 
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
+.. code-block:: typescript
+   :emphasize-lines: 2
 
-  .. tab:: Nested Objects
-     :tabid: nested-objects
+   interface RecursivePet {
+      pet?: RecursivePet;
+      name: string;
+      age: number;
+   }
 
-     The following code snippet raises a type error:
+Recursive Types and Dot Notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
+The {+driver-short+} cannot provide type safety within nested instances of
+recursive types referenced through **dot notation**. Dot notation is a property
+access syntax for navigating BSON objects.
 
-     This is the error:
+For example, the following code snippet references a nested instance of the
```

Comment:
"For example" seems unnecessary and disconnected with the previous sentence (definition of dot syntax). Usually, you use "for example" to describe a concept from the previous sentence.

```suggestion
The following code snippet references a nested instance of the
```